src/plugins/hetero/src/plugin.cpp

```diff
@@ -24,18 +24,104 @@
+                                                                       subgraph._parameters,
+                                                                       model_name + "_" + std::to_string(i));
+        }
+
+        return {mapping_info, compiled_submodels};
+    }
+
+    // Restore properties in order to pass "device priorities" together
+    // with devices properties
+    auto full_properties = config.get_hetero_properties();
+    for (const auto& [device, props] : config.get_device_properties()) {
+        full_properties[device] = props;
+    }
+
+    auto cloned_model = model->clone();
+    std::tie(query_model_result, mapping_info) = query_model_update(cloned_model, full_properties, true);
+
+    ov::hetero::op::DeviceSubgraphVector ordered_subgraphs;
+    for (const auto& op : cloned_model->get_ordered_ops()) {
+        if (const auto& subgraph = ov::as_type_ptr<ov::hetero::op::DeviceSubgraph>(op)) {
+            ordered_subgraphs.push_back(subgraph);
+        } else {
+            OPENVINO_ASSERT(ov::op::util::is_output(op) || ov::op::util::is_parameter(op) || ov::op::util::is_sink(op),
+                            "Unexpected node type found in model after query_model_update()");
+        }
+    }
+
+    compiled_submodels.resize(ordered_subgraphs.size());
+    for (size_t i = 0; i < ordered_subgraphs.size(); ++i) {
+        compiled_submodels[i].first = ordered_subgraphs[i]->get_affinity();
+        compiled_submodels[i].second = ordered_subgraphs[i]->get_function();
+    }
+
+    return {mapping_info, compiled_submodels};
+}
+
 std::shared_ptr<ov::ICompiledModel> ov::hetero::Plugin::compile_model(const std::shared_ptr<const ov::Model>& model,
                                                                       const ov::AnyMap& properties) const {
     OV_ITT_SCOPED_TASK(itt::domains::Hetero, "Plugin::compile_model");
 
     auto config = Configuration{properties, m_cfg};
-    auto compiled_model = std::make_shared<CompiledModel>(model->clone(), shared_from_this(), config);
-    return compiled_model;
+    auto cloned_model = model->clone();
+    SubgraphsMappingInfo mapping_info;
+    std::vector<ov::hetero::SubmoduleInfo> compiled_submodels;
+    std::tie(mapping_info, compiled_submodels) = split_graph(cloned_model, config);
+    ov::hetero::RemoteContext::Ptr remote_context;
+    std::shared_ptr<ov::hetero::CompiledModel> compiled_model;
+    try {
+        std::map<std::string, ov::SoPtr<ov::IRemoteContext>> contexts_map;
+        for (const auto& [device_name, _] : compiled_submodels) {
+            contexts_map.insert({device_name, get_core()->get_default_context(device_name)});
+        }
+        remote_context = std::make_shared<ov::hetero::RemoteContext>(std::move(contexts_map));
+    } catch (const ov::Exception&) {
+    }
+    return std::make_shared<CompiledModel>(cloned_model,
+                                           compiled_submodels,
```

Comment:
considering change the name? compiled*** is confusing

Reply:
Okay, will change the name.